contracts/abstract/AccessControlledAndUpgradeable.sol

```diff
@@ -18,6 +18,10 @@ abstract contract AccessControlledAndUpgradeable is Initializable, AccessControl
   bytes32 public constant UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
   /// @notice Role for the single entity that has admin control over the contract.
   bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+  /// @notice Role for the single entity that has important EMERGENCY mitigation actions.
+  bytes32 public constant EMERGENCY_ROLE = keccak256("EMERGENCY_ROLE");
+  /// @notice Role for the single entity that has ability to change some minor parameters in code.
+  bytes32 public constant MINOR_ADMIN_ROLE = keccak256("MINOR_ADMIN_ROLE");
```

Comment:
Added 2 more roles fore more fine grained access control.

We thought that just `admin` was too course. And we intend to use a timelock on these functions - eg. it makes sense that Emergency actions have a very short/no timelock, but shouldn't be able to do anything that affects user funds.